scenarios/examples/util.py

```diff
@@ -0,0 +1,427 @@
+import json
+from dataclasses import dataclass
+from secrets import randbelow
+from typing import Any, Dict, List, Mapping, Optional, Tuple, Type, Union
+from uuid import uuid4
+import time
+
+import docker
+from docker.models.containers import Container
+
+from acapy_controller import Controller
+from acapy_controller.controller import Minimal, MinType
+from acapy_controller.logging import logging_to_stdout
+from acapy_controller.models import (
+    CreateWalletResponse,
+    V20CredExRecordIndy,
+    V20PresExRecord,
+    V20PresExRecordList,
+)
+
+
+# docker utilities:
+def healthy(container: Container) -> bool:
+    """Check if container is healthy."""
+    inspect_results = container.attrs
+    return inspect_results["State"]["Running"] and inspect_results["State"]["Health"]["Status"] == "healthy"
+
+
+def unhealthy(container: Container) -> bool:
+    """Check if container is unhealthy."""
+    inspect_results = container.attrs
+    return not inspect_results["State"]["Running"]
+
+
+def wait_until_healthy(client, container_id: str, attempts: int = 350, is_healthy=True):
+    """Wait until container is healthy."""
+    container = client.containers.get(container_id)
+    print((container.name, container.status))
+    for _ in range(attempts):
+        if (is_healthy and healthy(container)) or unhealthy(container):
```

Comment:
I gotta ask... "healthy" or "unhealthy" is fine??

Reply:
"unhealthy" means we're shutting down the agent and we're checking that it's actually down (we let it shutdown cleanly).

Reply:
Sounds good.